src/queries.jl

```diff
@@ -259,15 +261,26 @@ for (tagsymbol, tagvariant) in pairs(tag_types)
         argssig_wild = [:($a::$t) for (a,t) in zip(args, sig_wild)]
         wild_checks = [:(isa($(args[i]),Wildcard) || $(args[i])(tag[$i])) for i in idx]
         nonwild_checks = [:(tag[$i]==$(args[i])) for i in complement_idx]
-        newmethod_reg = quote function query(reg::Register, $(argssig_wild...), ::Val{allB}=Val{false}(); locked::Union{Nothing,Bool}=nothing, assigned::Union{Nothing,Bool}=nothing) where {allB}
+        newmethod_reg = quote function query(reg::Register, $(argssig_wild...), ::Val{allB}=Val{false}(), ::Val{fifo}=Val{true}(); locked::Union{Nothing,Bool}=nothing, assigned::Union{Nothing,Bool}=nothing) where {allB, fifo}
             res = NamedTuple{(:slot, :tag), Tuple{RegRef, Tag}}[]
             for (reg_idx, tags) in enumerate(reg.tags)
                 slot = reg[reg_idx]
-                for tag in tags
-                    if isvariant(tag, ($(tagsymbol,))[1]) # a weird workaround for interpolating a symbol as a symbol
-                        (_nothingor(locked, islocked(slot)) && _nothingor(assigned, isassigned(slot))) || continue
-                        if _all($(nonwild_checks...)) && _all($(wild_checks...))
-                            allB ? push!(res, (;slot, tag)) : return (;slot, tag)
+                if fifo
+                    for tag in reverse(tags)
+                        if isvariant(tag, ($(tagsymbol,))[1]) # a weird workaround for interpolating a symbol as a symbol
+                            (_nothingor(locked, islocked(slot)) && _nothingor(assigned, isassigned(slot))) || continue
+                            if _all($(nonwild_checks...)) && _all($(wild_checks...))
+                                allB ? push!(res, (;slot, tag)) : return (;slot, tag)
+                            end
+                        end
+                    end
+                else
+                    for tag in tags
+                        if isvariant(tag, ($(tagsymbol,))[1]) # a weird workaround for interpolating a symbol as a symbol
+                            (_nothingor(locked, islocked(slot)) && _nothingor(assigned, isassigned(slot))) || continue
+                            if _all($(nonwild_checks...)) && _all($(wild_checks...))
+                                allB ? push!(res, (;slot, tag)) : return (;slot, tag)
+                            end
                         end
                     end
                 end
```

Comment:
```suggestion
                for tag in (filo ? reverse(tags) : tags) 
                    if isvariant(tag, ($(tagsymbol,))[1]) # a weird workaround for interpolating a symbol as a symbol
                        (_nothingor(locked, islocked(slot)) && _nothingor(assigned, isassigned(slot))) || continue
                        if _all($(nonwild_checks...)) && _all($(wild_checks...))
                            allB ? push!(res, (;slot, tag)) : return (;slot, tag)
                        end
                    end
                end
```

This way we avoid dangerous code repetition.

Reply:
and then we want to be careful to do `reverse(res)` in the case of `allB == true and filo == true` at the end.

Reply:
We wouldn't need to do `reverse(res)` here, because based on `filo` we would already be looking serially backwards or forwards. We could probably remove the conditional in the `for` statement above and just do `reverse(res)` that should also work